monolith/views/test/test_auth.py

```diff
@@ -0,0 +1,82 @@
+from monolith.views.test.TestHelper import TestHelper
+from sqlalchemy.exc import IntegrityError
+from monolith.database import db, User
+from flask_login import current_user
+
+class TestAuth(TestHelper):
+
+
+    def test_login(self):
+        
+        # error:wrong pass
+        reply = self._login("example@example.com", "42")
+        self.assert_context("notlogged", True)
+        with self.context:
+            q = User.query.filter_by(email="example@example.com")
+            self.assertNotEqual(q.first(), current_user)
+
+        # success: logged in
+        reply = self._login("example@example.com", "admin")
+        self.assertEqual(reply.status_code, 302)
+        with self.context:
+            q = User.query.filter_by(email="example@example.com").first()
+            #self.assertEqual(current_user.id, q.id) # current_user anonymous?
+        
+    def test_login_redir(self):
+    
+        # success: logged in, redirected
+        reply = self._login("example@example.com", "admin", True)
+        self.assertEqual(reply.status_code, 200)
+        self.assert_template_used("home.html")
+        
+    def test_logout(self):
+    
+        # error:not logged in
+        reply = self._logout()
+        self.assertEqual(reply.status_code, 401)
+        
+        # success:logged out, no redir
+        reply = self._login("example@example.com", "admin")
+        reply = self._logout()
+        self.assertEqual(reply.status_code, 302)
+        
+        # success:logged out, redirected
+        reply = self._login("example@example.com", "admin")
+        reply = self._logout(True)
+        self.assertEqual(reply.status_code, 200)
+        self.assert_template_used("login.html")
+        
+    def test_signup(self):
+    
+        # error:same email
+        reply = self._signup('example@example.com', 'admin', 'admin', 'giacobbe', '01/04/1006')
+        self.assertRaises(IntegrityError)
```

Comment:
From my understanding, to use assertRaises() you have to pass the function that is to be checked and the arguments to that function or put it in a with statement.

See https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaises.

Reply:
I think the with statement is optional, if you wish to test the exception received. In this way I only check that we receive the correct exception

Reply:
assertRaises(exc, fun, *args, **kwds)

Reply:
Ok you'r right, I'll just remove it for now cause the exception gets caught anyway

Reply:
You can just do self.assertRaises(IntegrityError, self._signup, 'example@example.com', 'admin', 'admin', 'giacobbe', '01/04/1006').

Otherwise I think the testing script crashes instead of marking the test as failing

Reply:
ok nvm, as we discussed offline we are catching that exception inside the route handler, so we can't see it from outside.